### PR TITLE
use bcrypt instead of py-bcrypt for supporting Windows + python3

### PR DESCRIPTION
**Pull request by growingdever**
I met msvc++ 10.0 error while developing on windows with python3 because of py-bcrypt package.
To solve that problem, I found bcrypt pacakge that doesn't cause error and replace py-bcrypt to bcrypt.

### CODE REVIEW

File: test_bcrypt.py [hunk 1]
```diff
@@ -13,11 +13,11 @@ def setUp(self):
 
     def test_is_string(self):
         pw_hash = self.bcrypt.generate_password_hash('secret')
-        self.assertTrue(isinstance(pw_hash, str))
+        self.assertTrue(isinstance(pw_hash, bytes))
```
Comment (maxcountryman) [inline test_bcrypt.py]: Isn't this going to break existing applications?

**Comment by growingdever** [comment]
I changed result as str

**Comment by joshfriend** [comment]
+1 for switching to the `bcrypt` package distributed by PYCA. It uses the newer `2b` hashes by default instead of the older and possibly insecure `2a` hashes.

Unsure about the other changes in this PR though, everything works fine and tests pass without it.

**Comment by maxcountryman** [comment]
@joshfriend okay, agree with your analysis here. However I'm also unsure about the other changes in this commit which is why I haven't merged it in yet.

File: flask_bcrypt.py [hunk 2]
```diff
@@ -155,8 +155,19 @@ def generate_password_hash(self, password, rounds=None):
             password = password.encode('u8')
         elif PYVER >= 3 and isinstance(password, bytes):
             password = password.decode('utf-8')
-        password = str(password)
-        return bcrypt.hashpw(password, bcrypt.gensalt(rounds))
+
+        if PYVER < 3:
+            password = str(password)
```
Comment (joshfriend) [inline flask_bcrypt.py]: This will raise `UnicodeDecodeError` in python2 if `password` is a unicode string that contains non-ascii characters.

**Comment by maxcountryman** [comment]
If everyone is satisfied with 6b9a04a22076b6c4962bcd080296e7e5c3aad3d8 I'll publish a release to PyPI.

**Comment by joshfriend** [comment]
@maxcountryman LGTM :+1:

**Comment by maxcountryman** [comment]
`0.7.0` is now on PyPI.

**Comment by smewp** [comment]
Just some feedback, in upgrading from 0.6.2, this switch to 0.7.1 was totally painful in that I could not eliminate a "Invalid salt" error when trying to login from database in my app.  After working on it half a day I had to reinstall 0.6.2, which still works fine.  Here is a hash saved to database that works good with 0.6.2     `$2a$12$coIF/8pB06w4b3nQG/cLfOT8Y33TtdlYjFNfAzBolRyPCcdeSC6qK` and here is hash with the 0.7.1 `\x24326224313224524d4f614246376179433269524747387a7131667a2e7146487264377039705442675965745650764268373662737a58366676794b` which it just isnt working, just keep getting "Invalid salt".

**Comment by asonnenschein** [comment]
@maxcountryman I'm seeing the same peculiarity that @smewp is seeing.  Version 0.6.2 works great, but version 0.7.1 throws 'Invalid salt' errors every time I run the check_password_hash function.

**Comment by raylu** [comment]
@joshfriend @maxcountryman
> +1 for switching to the bcrypt package distributed by PYCA. It uses the newer 2b hashes by default instead of the older and possibly insecure 2a hashes.

AFAICT, there is no security concern with `$2a$` hashes on non-BSD systems. The vulnerability was in OpenBSD: https://marc.info/?l=openbsd-misc&m=139320023202696
and they updated the version number for their fix

as proof, with py-bcrypt:
```python
>>> s = bcrypt.gensalt()
>>> s
'$2a$12$wXKjnWImM/JumxlcalGxZ.'
>>> bcrypt.hashpw('OrpheanBeholderScryDoubt', s)
'$2a$12$wXKjnWImM/JumxlcalGxZ.TIp254ON.OsAPI3uD1sjKKl7M6f.T9C'

>>> bcrypt.hashpw('OrpheanBeholderScryDoubt', '$2a$12$K7qlMy5UjNRJH9Ws0ESyEu') # salt from below, replacing b with a
'$2a$12$K7qlMy5UjNRJH9Ws0ESyEug3d4FE918Dg5efeOsN6AP0ZY0Ts0wW6'
```
with bcrypt:
```python
>>> bcrypt.hashpw('OrpheanBeholderScryDoubt', '$2a$12$wXKjnWImM/JumxlcalGxZ.') # salt from above
'$2a$12$wXKjnWImM/JumxlcalGxZ.TIp254ON.OsAPI3uD1sjKKl7M6f.T9C'

>>> s = bcrypt.gensalt()
>>> s
'$2b$12$K7qlMy5UjNRJH9Ws0ESyEu'
>>> bcrypt.hashpw('OrpheanBeholderScryDoubt', s)
'$2b$12$K7qlMy5UjNRJH9Ws0ESyEug3d4FE918Dg5efeOsN6AP0ZY0Ts0wW6'
```

**Pull request closed without merging**